packages/flutter/test/widgets/sensitive_content_error_handling_test.dart

```diff
@@ -0,0 +1,141 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/src/widgets/sensitive_content.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import 'sensitive_content_utils.dart';
+
+void main() {
+  final SensitiveContentHost sensitiveContentHost = SensitiveContentHost.instance;
+
+  tearDown(() {
+    TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+      SystemChannels.sensitiveContent,
+      null,
+    );
+  });
+
+  testWidgets(
+    'when SensitiveContentService.setContentSensitivity fails, SensitiveContentHost.register throws FlutterError but still updates calculatedContentSensitivity',
+    (WidgetTester tester) async {
+      int setContentSensitivityCall = 0;
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
```

Comment:
I think after doing this you also should remove it:

```dart
      addTearDown(
        () => tester.binding.defaultBinaryMessenger.setMockMethodCallHandler(
          SystemChannels.sensitiveContent,
          null,
        ),
      );
```

Reply:
Won't the `tearDown` do this?

Reply:
Ah sorry I totally missed that above, this is good as-is :+1: